server/service/src/sync/sync_serde.rs

```diff
@@ -16,12 +16,22 @@ pub fn empty_str_as_option_string<'de, D: Deserializer<'de>>(
 pub fn empty_str_as_option<'de, T: Deserialize<'de>, D: Deserializer<'de>>(
     d: D,
 ) -> Result<Option<T>, D::Error> {
-    let s: Option<String> = empty_str_as_option_string(d)?;
-
-    let Some(s) = s else { return Ok(None) };
-
-    let str_d: StrDeserializer<D::Error> = s.as_str().into_deserializer();
-    Ok(Some(T::deserialize(str_d)?))
+    let value: Value = Value::deserialize(d)?;
+    match value {
+        Value::String(ref s) if s.is_empty() => Ok(None),
+        Value::Null => Ok(None),
+        _ => {
+            let value_str = format!("{:?}", value);
+            let t: T = T::deserialize(value).map_err(|e| {
+                Error::custom(format!(
+                    "Failed to deserialize value: {}. Error: {}",
+                    value_str,
+                    format_error(&e)
+                ))
+            })?;
+            Ok(Some(t))
+        }
+    }
```

Comment:
This isn't part of this issue - but I was wanting to use this deseriliaser to deserialise my optional number fields which could be an empty string. 

I realised they shouldn't ever be an empty string, but during this I noticed that this deserialiser will panic on a non string value which is confusing given its name. It was acting no differently than empty_str_as_option_string.

So have added this change so it is safe on type T deserilisation + tests.

As a side note - I think all of these deserialisers should have comprehensive test cases and have added issue [here](https://github.com/msupply-foundation/open-msupply/issues/8270)

Reply:
James pointed out that it wouldn't be the case and numbers would be 0 in OG. However to me the name implies that it is safe for generic type T, and not just strings, so I was wanting to make it safe.

Maybe it isn't necessary? I was not confident of type security coming from OG so wasn't sure if floats, etc could be returned as en empty string